web/src/main/java/org/mskcc/cbio/oncokb/controller/EvidenceController.java

```diff
@@ -276,48 +278,68 @@ private Boolean isEmptyEvidence(Evidence queryEvidence) {
                 }
             }
         }
-        // cancerTypes and subTypes are required to be the same length
+        // if cancerTypes and subTypes are not the same length, return null
         if(cancerTypes.size() != subTypes.size()) {
             return null;
         }
-        
-        
-
+        EvidenceBo evidenceBo = ApplicationContextSingleton.getEvidenceBo();
+        List<Evidence> evidences = evidenceBo.findEvidenceByUUIDs(Collections.singletonList(uuid));
+               
+        EvidenceType evidenceType = queryEvidence.getEvidenceType();
+        // Three special evidences update, which are mutation name change, tumor name change and treatment name change
+        if(evidenceType == null) {
+           // If evidences not exist, return empty list
+           if(evidences.isEmpty())return new ArrayList<Evidence>(); 
+           String specialChangeType = specialTypeCheck(queryEvidence);
+           if(specialChangeType.equals("MUTATION_NAME_CHANGE")) {
```

Comment:
Because specialTypeCheck return null if the queryEvidence not in three special changes, you need to do null check here, otherwise it will throw exception. I just you return empty string instead.